app/src/main/java/com/ucm/informatica/spread/Activities/MainTabActivity.java

```diff
@@ -108,20 +108,25 @@ public void initView(){
     }
 
     @Override
-    public void loadDataEventSmartContract(String title, String description, String latitude, String longitude, String dataTime) {
-        dataEventSmartContractList.add(new Event(this, title,description, latitude, longitude, dataTime));
+    public void loadDataAlertSmartContract(String title, String description, String latitude, String longitude, String dataTime) {
+        dataAlertSmartContractList.add(new Alert(this, title,description, latitude, longitude, dataTime));
     }
 
     @Override
-    public void loadDataPosterSmartContract(String title, String description, String latitude, String longitude, String dataTime, byte[] image) {
-        dataPosterSmartContractList.add(new Poster(this, title,description, latitude, longitude, dataTime, image));
+    public void loadDataPosterIPFS(Poster poster) {
```

Comment:
whats going on here ? 

1.  Poster info (title, descrip, lat, longi ...) will be represented as a JSON, easier to store as a single string, then it will be saved into IPFS.
2. IPFS will return a hash for this data.
3. We get this hash and we save it into Ethereum via PosterContract.

NOTE : IPFS and Ethereum nodes run on Infura .